maplibre_gl_web/lib/src/maplibre_web_gl_platform.dart

```diff
@@ -988,24 +989,47 @@ class MapLibreMapController extends MapLibrePlatform
 
     if (enableInteraction) {
       _interactiveFeatureLayerIds.add(layerId);
-      if (layerType == "fill") {
-        _map.on('mousemove', layerId, _onMouseEnterFeature);
-      } else {
-        _map.on('mouseenter', layerId, _onMouseEnterFeature);
-      }
+      _map.on('mouseenter', layerId, _onMouseEnterFeature);
+      _map.on('mousemove', layerId, _onMouseMoveInFeature);
       _map.on('mouseleave', layerId, _onMouseLeaveFeature);
       if (_dragEnabled) _map.on('mousedown', layerId, _onMouseDown);
     }
   }
 
-  void _onMouseEnterFeature(_) {
+  void _onMouseEnterFeature(Event e) {
     if (_draggedFeatureId == null) {
       _map.getCanvas().style.cursor = 'pointer';
     }
+    final featureId = e.features.first.id;
```

Comment:
Empty features will throw an exception. If they are empty, we return to avoid this
```suggestion
    if (e.features.isEmpty) return;
    final featureId = e.features.first.id;
```